src/Http/Resources/SettingsCollection.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace EscolaLms\Settings\Http\Resources;
+
+use Illuminate\Http\Resources\Json\ResourceCollection;
+
+class SettingsCollection extends ResourceCollection
+{
+    /**
+     * Transform the resource collection into an array.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function toArray($request)
+    {
+        $data = [];
+
+        foreach ($this->collection as $value) {
+            switch($value->type) {
+                case "config":
+                    $data[$value->group][$value->key] = config($value->value);
+                    break;
+                case "json":
+                    $data[$value->group][$value->key] = json_decode($value->value);
+                    break;
+                default:
+                $data[$value->group][$value->key] = $value->value;
```

Comment:
Wouldn't it be possible to use custom Caster in Model for casting this value to correct type?

https://laravel.com/docs/8.x/eloquent-mutators#custom-casts